### ISSUE TITLE
Inconsistency with paper in connection from vstack to hstack

### ISSUE THREAD

**Issue by kkleidal** [anantzoid/Conditional-PixelCNN-decoder#6]
https://github.com/anantzoid/Conditional-PixelCNN-decoder/blob/9a5c9a3df2c58100cf5e3600392e67db8ac7a59e/models.py#L30-L34

This has the connection from the vstack to the hstack occuring after the gate, but in Figure 2 of the paper, it occurs after the masked convolution but before splitting the channels and gating.  Any particular reason for this change?

**Comment by falconjhc** [comment]
Yeah I also found the issue. Is it the official code written by the original paper author?

**Comment by kkleidal** [comment]
No it is not. I was exploring using PixelCNN for a class project and never fully got it working which is why I added the disclaimer on the README. I haven’t found the time to get around to fixing it yet.

You’ll notice that there are quite a few forks of my project though; perhaps one of them fixed the bugs and have a working version you could use.

Sorry for the inconvenience and best of luck!

--
Ken Leidal

**Comment by kkleidal** [comment]
Sidenote: Sorry for my previous comment (now deleted). I didn't read the email closely enough and thought it was in response to a similarly named repo I made.

I don't think this is the official implementation. I did try to correct the issue in #7 which has since been merged.

**Comment by yuffon** [comment]
In the original paper, the link from vstack to hstack is before gate.
This code is different.

**Comment by Raaaaaaat** [comment]
I'm new to this paper, and I found that the h_stack is different with the original paper.
the code is"_[filter_size if full_horizontal else 1, filter_size, conf.f_map]_"
and para "full_horizontal" is set as TRUE
Cuz the size of mask is changging from 7x7 to 3x3, and perhaps the size of mask should be 1x7/1x3?
I think the size of horizontal stack is changging from 1x3 to 1x7, but maybe I understand the wrong way, anyone can explain why the kernel changes from 7x7 to 3x3 and why it is 7x7 instead of 1x7?
I'm  so confused, plzplzplz help me out, I really appreciate your help! THX!!!!
